bincapz.go

```diff
@@ -32,6 +32,7 @@ func main() {
 	statsFlag := flag.Bool("stats", false, "show statistics about the scan")
 	verboseFlag := flag.Bool("verbose", false, "emit verbose logging messages to stderr")
 	ociFlag := flag.Bool("oci", false, "scan an OCI image")
+	archiveFlag := flag.Bool("archive", false, "scan an archive (.zip, .tar, .jar, etc.)")
```

Comment:
rather than introducing another flag, can we process archives transparently?

Ideally, I'd like bincapz to be able to process archives found in a recursive directory search, but this adds complication - so IMHO it's best to defer that to a second PR.

If it helps, there is already a `programKind` function that can be extended.

Reply:
Addressed this in [`c72990f` (#144)](https://github.com/chainguard-dev/bincapz/pull/144/commits/c72990f25f276ca936c4d764f7886dc8cac3cf0c) and [`3a0f288` (#144)](https://github.com/chainguard-dev/bincapz/pull/144/commits/3a0f288429c28797a5b00ab9aad65f5a7bd54302).

I added functionality to extract the file extension and used a separate map for the archive types to directly check whether the path is an archive or not.